add website post (#42)
OBF/OBF.github.io: content/posts/2025-03-04-new-website.md

---
author: Bastian Greshake Tzovaras
category:
  - community
  - obf 
date: "2025-03-04T18:23:00+00:00"
tag:
  - obf
  - internal
 
title: "Welcome to the new OBF website!"
---

![the hugo logo](/img/2025-03-04-hugo.svg)

Over the past few weeks, we have been working behind the scenes to migrate our website, resulting in the page that you currently have in front of you!

> Our pages are now generated by the [_Go_-based, open source static site generator _Hugo_](https://gohugo.io/).

Reviewer: ```suggestion
Our pages are now generated by the [_Go_-based, open source static site generator _Hugo_](https://gohugo.io/), and are stored on [GitHub](https://github.com/OBF/OBF.github.io).
```

> All old URLs should continue working as before, but **if you run into any missing pages, please let us know through any of the channels linked in the page footer**.

Reviewer: ```suggestion
All old URLs should continue working as before, but **if you run into any missing pages or other issues, please let us know by [submitting an issue](https://github.com/OBF/OBF.github.io/issues/new) or contacting us via the channels linked in the page footer**.
```

Until recently, the OBF websites and blog ran on _Wordpress_, the most common, PHP-based Content Management System, which overall has served us well over the years (and which itself was a migration from the previous _MediaWiki_-based setup).
Beyond the [recent governance issues with Wordpress](https://www.theverge.com/2024/9/27/24256361/wordpress-wp-engine-drama-explained-matt-mullenweg), we also decided to move towards a static site system due to being easier to maintain in the long-term.

Keeping up with the security patches for WP itself, the plugins we used and keeping it compatible with our custom theme required a lot of effort.

> As a small, all-volunteer-run organisation, we decided that it's both a better use of our time but also more sustainable in the long-term to use a simpler static site generator.

Reviewer: ```suggestion
As a small, all-volunteer-run organisation, we decided that it's both a better use of our time and more sustainable in the long-term to use a simpler static site generator.
```

> As ultimately the output is just a collection of HTML files that can easily be moved between hosts and are mostly self-contained.

Reviewer: ```suggestion
Ultimately, the output of the static site generator is just a collection of HTML files that can easily be moved between hosts and are mostly self-contained.
```

For our use-case, _Hugo_ and its ecosystem have the benefit that [_wp2hugo_](https://github.com/ashishb/wp2hugo) provides an easy way to migrate an existing blog from Wordpress to Hugo!

> Based on the _XML dump_ that Wordpress can generate, it can create all the markdown files and even get the necessary static files like images in the right place.

Reviewer: ```suggestion
Using an _XML dump_ that Wordpress can generate, Hugo can create all the markdown files and even get the necessary static files like images in the right place.
```
